Description: Wider support for varying web root of more OS flavours.
Author: Dave Walker (Daviey) <DaveWalker@ubuntu.com>, Chuck Short <zulcss@ubuntu.com>
Forwarded: no
--- a/Makefile
+++ b/Makefile
@@ -4,6 +4,7 @@
 
 prefix=devinstall
 statepath=/tmp/cobbler_settings/$(prefix)
+www-user=www-data
 
 all: clean build
 
@@ -37,10 +38,10 @@
 
 install: build
 	python setup.py install -f
-	chown -R apache /usr/share/cobbler/web
+	chown -R $(www-user) /usr/share/cobbler/web
 
 debinstall:
-	python setup.py install -f --root $(DESTDIR)
+	python setup.py install -f --root $(DESTDIR) --install-layout=deb
 
 devinstall:
 	-rm -rf /usr/share/cobbler
@@ -72,7 +73,7 @@
 	cp $(statepath)/dhcp.template /etc/cobbler/dhcp.template
 	cp $(statepath)/rsync.template /etc/cobbler/rsync.template
 	find /var/lib/cobbler/triggers | xargs chmod +x
-	chown -R apache /var/www/cobbler
+	chown -R $(www-user) /var/www/cobbler
 	chmod -R +x /var/www/cobbler/web
 	chmod -R +x /var/www/cobbler/svc
 	rm -rf $(statepath)
--- a/build_backup/Makefile
+++ b/build_backup/Makefile
@@ -4,6 +4,7 @@
 
 prefix=devinstall
 statepath=/tmp/cobbler_settings/$(prefix)
+www-user=www-data
 
 all: clean build
 
@@ -42,7 +43,7 @@
 
 install: build manpage
 	python setup.py install -f
-	chown -R apache /usr/share/cobbler/web
+	chown -R $(www-user) /usr/share/cobbler/web
 
 debinstall: manpage
 	python setup.py install -f --root $(DESTDIR)
@@ -77,7 +78,7 @@
 	cp $(statepath)/dhcp.template /etc/cobbler/dhcp.template
 	cp $(statepath)/rsync.template /etc/cobbler/rsync.template
 	find /var/lib/cobbler/triggers | xargs chmod +x
-	chown -R apache /var/www/cobbler
+	chown -R $(www-user) /var/www/cobbler
 	chmod -R +x /var/www/cobbler/web
 	chmod -R +x /var/www/cobbler/svc
 	rm -rf $(statepath)
--- a/setup.py
+++ b/setup.py
@@ -128,7 +128,13 @@
     libpath     = "/var/lib/cobbler/"
     logpath     = "/var/log/"
 
-    webroot     = "/var/www/"
+    if os.path.exists("/etc/Suse-release"):
+        webroot = "/srv/www"
+    elif os.path.exists("/etc/debian-version"):
+        webroot = "/usr/share/cobbler/webroot"
+    else:
+        webroot     = "/var/www/"
+
     if os.path.exists("/etc/SuSE-release"):
         webconfig  = "/etc/apache2/conf.d"
     elif os.path.exists("/etc/debian_version"):
@@ -173,8 +179,6 @@
             ("/usr/sbin", ["scripts/tftpd.py"]),
 
             ("%s" % webconfig,              ["config/cobbler.conf"]),
-            ("%s" % webconfig,              ["config/cobbler_web.conf"]),
-            ("%s" % initpath,               ["config/cobblerd"]),
             ("%s" % docpath,                ["docs/*.gz"]),
             ("installer_templates",         ["installer_templates/*"]),
             ("%skickstarts" % libpath,      ["kickstarts/*"]),
